doc/04-Security.md

```diff
@@ -0,0 +1,71 @@
+# Security
+
+Icinga for Kubernetes allows users to show different Kubernetes resources. Users may be restricted to a specific set of
+resources, by use of **permissions** and **restrictions**.
+
+## Permissions
+
+> If a role [limits users](#filters) to a specific set of results, the
+> permissions or refusals of the very same role only apply to these results.
+
+| Name                                     | Allow...                         |
+|------------------------------------------|----------------------------------|
+| kubernetes/config-maps/show              | to show config maps              |
+| kubernetes/cron-jobs/show                | to show cron jobs                |
+| kubernetes/daemon-sets/show              | to show daemon sets              |
+| kubernetes/deployments/show              | to show deployments              |
+| kubernetes/events/show                   | to show events                   |
+| kubernetes/ingresses/show                | to show ingresses                |
+| kubernetes/jobs/show                     | to show jobs                     |
+| kubernetes/nodes/show                    | to show nodes                    |
+| kubernetes/persistent-volume-claims/show | to show persistent volume claims |
+| kubernetes/persistent-volumes/show       | to show persistent volumes       |
+| kubernetes/pods/show                     | to show pods                     |
+| kubernetes/replica-sets/show             | to show replica sets             |
+| kubernetes/secrets/show                  | to show secrets                  |
+| kubernetes/services/show                 | to show services                 |
+| kubernetes/stateful-sets/show            | to show stateful sets            |
+| kubernetes/yaml/show                     | to show yaml                     |
+
+## Restrictions
+
+### Filters
+
+Filters limit users to a specific set of results.
+
+> **Note:**
+>
+> Filters from multiple roles will widen available access.
+
+| Name                        | Description                                                       |
+|-----------------------------|-------------------------------------------------------------------|
+| kubernetes/filter/resources | Restrict access to the Kubernetes resources that match the filter |
+
+`kubernetes/filter/resources` will only allow users to access matching Kubernetes resources. This applies to all
+resources.
+
+Allowed columns are namespace and name.
+
+> **Note:**
+>
+> Nodes, namespaces and persistent volumes do not belong to a namespace, therefore only the name is available for
+> filtering.
+
+## Restricted Permissions:
+
+If a user has permission to show one resource but lacks permissions for another resource that is dependent on or related
```

Comment:
Restricted permissions is the combination of permission(s) and restriction(s), i.e. if I have two roles where role a allows to list pods and has defined a certain restriction and role b allows to list deployments and has specified a certain restriction, I see pods that are restricted by role a and I see deployments that are restricted by role b. We have introduced something similar in Icinga DB. Previously, with the functionality as it exists in the monitoring module for example, I would see pods and deployments that match both role a and role b restrictions.

Though, I think the explanation of the permissions is good and would keep it, but move it to the section on permissions.